internal/common/serve/static.go

```diff
@@ -9,6 +10,27 @@ import (
 	"github.com/armadaproject/armada/internal/common/logging"
 )
 
+// dirWithIndexFallback is a http.FileSystem that serves the index.html file at
+// the root of dir if the requested file is not found. This behavior differs
+// from http.Dir, which only forwards requests for /a/ to /a/index.html; we need
+// to serve the index.html file at the root of dir if the requested file is not
+// found, so that the frontend can handle routing in those cases.
+type dirWithIndexFallback struct {
+	dir http.Dir
+}
+
+func CreateDirWithIndexFallback(path string) http.FileSystem {
+	return dirWithIndexFallback{http.Dir(path)}
+}
+
+func (d dirWithIndexFallback) Open(name string) (http.File, error) {
+	file, err := d.dir.Open(name)
+	if errors.Is(err, fs.ErrNotExist) {
```

Comment:
This is essentially the code that I overzealously removed in #3202, expect that I have replaced `err != nil` by `errors.Is(err, fs.ErrNotExist)` here.